### ISSUE TITLE
[Improvement] Avoid removeResources for multiple times

### ISSUE THREAD

**Issue by xianjingfeng** [apache/incubator-uniffle#456]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

If `removeResources` take to much time. NPE will be thrown as follows.

```shell
[ERROR] 2022-12-30 00:43:28,403 clearResourceThread ShuffleTaskManager lambda$new$3 - Exception happened when clear resource for expired application
java.lang.NullPointerException
        at org.apache.uniffle.server.ShuffleTaskManager.removeResources(ShuffleTaskManager.java:479)
        at org.apache.uniffle.server.ShuffleTaskManager.lambda$new$3(ShuffleTaskManager.java:130)
        at java.lang.Thread.run(Thread.java:745)

```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by zuston** [comment]
Nice catch, could you help fix this? If not, you could give some points to fix and mark it as `good first issue`.

**Comment by zuston** [comment]
Fixed by #459
